build/config/build.yaml

```diff
@@ -82,4 +87,61 @@ steps:
   dir: 'tests/'
   waitFor: ['DOCKER_STAGE', 'DOCKER_BUILD_MVN_GCLOUD']
 
+#
+# End of App Engine Integration tests
+
+#
+# GKE Integration tests
+#
+
+# Create a Docker image with the test application
+- name: 'gcr.io/cloud-builders/docker'
+  id: 'DOCKER_BUILD_TEST_APPLICATION'
+  args: ['build', '--tag=${_STAGING_IMAGE}-test-app', '--no-cache', 'tests/runtimes-common-testing/target/deploy']
```

Comment:
`mvn clean install` only cleans before the install. So, the `target` directory should still be around *after* it's done. However, if the steps run in parallel, then we have a problem with the clean being in the middle.

Anyway, why not just build the application container before running `JETTY_SMOKE_TEST`, and let `JETTY_SMOKE_TEST` do a clean because it may be passing different parameters to the build?